components/profile/components/EditProfile/index.js

```diff
@@ -27,27 +27,68 @@ const {
 const { plus } = ICONS;
 
 class EditProfile extends Component {
-    state = { 
-        fileList: [], 
-        location: null, 
-        locationLoading: false, 
-        previewImage: '', 
-        previewVisible: false, 
+    constructor(props){
+        super(props);
+
+        const { firstName, lastName, bio, highSchool, intendedUni, username } = props.user;
+
+        this.state = {
+            bio: bio || '',
+            fileList: [],
+            firstName: firstName || '',
+            highSchool: highSchool || '',
+            intendedUni: intendedUni || '',
+            lastName: lastName || '',
+            loading: false,
+            location: null,
+            locationLoading: false,
+            previewImage: '',
+            previewVisible: false,
+            username: username || '',
+        };
+
+    }
+
+    handleTextChange = e => {
+        const { name, value } = e.target;
+        this.setState({ [name]: value });
     }
 
+    handleSubmit = async () => {
+        const { updateUser } = this.props;
+        const { firstName, lastName, highSchool, intendedUni, username, bio } = this.state;
+        const data = { bio, firstName, highSchool, intendedUni, lastName,  username };
+        this.setState({ loading: true });   
+
+        updateUser(data);
+
+        this.setState({ loading: false }, () => {
+            message.success('Profile Update successfully');
+        });
+
+    };
+
     generateComponent = (name, label, extra) => {
         const { fileList, location, locationLoading, previewImage, previewVisible } = this.state;
         switch (name) {
-        case 'dateJoined':
         case 'firstName':
         case 'highSchool':
         case 'intendedUni':
         case 'lastName':
         case 'username':
-            return <Input name={name} />;
+            // eslint-disable-next-line react/destructuring-assignment
+            return <Input name={name} onChange={this.handleTextChange} value={this.state[name]} />;
```

Comment:
`this.state[name]` don't do this, there should never be a key on the state that is driven by a random value. you can go `this.state.name`

Reply:
I did that to avoid repetition. The values of [name] are basically from the switch case

Reply:
hmmm...i see, thi pattern might change after you move most of this state into redux